### PR TITLE
Make spacebar follow links

### PR DESCRIPTION
**Pull request by ali-kha**
while focused on a link (tabbing through the ui), pressing space will by default act as a "scroll down". This is inconsistent with pressing space on a button which will perform the action. So I changed all the links to also follow on space. If **not** focused on a link, space bar will still act as scroll.

### CODE REVIEW

File: src/client/src/utils/spaceKeyHandler.ts [hunk 1]
```diff
@@ -0,0 +1,22 @@
+import * as React from "react";
+import {KEY_EVENTS} from "./constants"
+
+// document.documentElement.addEventListener('keydown', (e) => {
```
Comment (SahilTara) [inline src/client/src/utils/spaceKeyHandler.ts]: Remove commented out code please

File: src/client/src/utils/spaceKeyHandler.ts [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+//   if (e.key === KEY_EVENTS.SPACE) e.preventDefault();
+// })
+
+const spaceKeyHandler = (e: React.KeyboardEvent<HTMLAnchorElement>) => {
+
+  let anchor = e.target as HTMLAnchorElement;
```
Comment (SahilTara) [inline src/client/src/utils/spaceKeyHandler.ts]: const here please.

File: src/client/src/utils/spaceKeyHandler.ts [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+
+  //if pressing space on an anchor, prevent page scroll.
+  anchor.onkeydown = (e) => {
```
Comment (SahilTara) [inline src/client/src/utils/spaceKeyHandler.ts]: Curious, does e.stopPropogation() not work?   Wondering why you're attaching an event listener inside the event listener. Feels a bit odd. Can you explain why?

File: src/client/src/components/Details/index.tsx [hunk 4]
```diff
@@ -92,7 +93,7 @@ const Details = ({
             role="button"
             tabIndex={0}
             onClick={handleBackClick}
-            onKeyDown={keyDownHandler}
+            onKeyUp={keyDownHandler}
```
Comment (streetcoder123) [inline src/client/src/components/Details/index.tsx]: - Why did you change from `onKeyDown` to `onKeyUp`?
- If we have to change it to `onKeyUp`, we should probably update the function to be called `keyUpHandler` instead.

File: src/client/src/containers/About/index.tsx [hunk 5]
```diff
@@ -38,6 +39,7 @@ const About = ({ versions, intl }: Props) => {
         <a
           className={styles.link}
           href="https://github.com/Microsoft/WebTemplateStudio"
+          onKeyUp={spaceKeyHandler}
```
Comment (streetcoder123) [inline src/client/src/containers/About/index.tsx]: Would it make sense to name this function as `keyUpHandler` so that later we can use the same function to handle other key press events as well (if required) using the switch/case statement?

File: src/client/src/utils/spaceKeyHandler.ts (hunk 3 above)
Comment (ali-kha) [reply src/client/src/utils/spaceKeyHandler.ts]: e.stopPropogation pretty much just "pops" the event bubble, but all event handlers left on the current element (the anchor in this case) will still occur, so that doesn't work, but to be safe I just tested it and it doesn't work. 

The reason I did the inner event listener is because that way the event listener would only be added to elements that we select (whichever elements we add this handler to). The more generic option would be to add an event listener to all a tags with something like `document.getElementsByTageName("a");` I just thought this would be more localized and cause less potential side effects.

File: src/client/src/components/Details/index.tsx (hunk 4 above)
Comment (ali-kha) [reply src/client/src/components/Details/index.tsx]: oh my bad, good catch, this wasn't something I was modifying but I mistook it as my code.

File: src/client/src/containers/About/index.tsx (hunk 5 above)
Comment (ali-kha) [reply src/client/src/containers/About/index.tsx]: yeah I didn't think about that but it would be better, changing that now

**Review by annielovesjs — DISMISSED** [review]
Nice! It works well

File: src/client/src/containers/Licenses/index.tsx [hunk 6]
```diff
@@ -38,7 +40,7 @@ const Licenses = ({
 }: Props) => {
   const LinkRenderer = (props: any) => {
     return (
-      <a href={props.href} className={styles.link}>
+      <a href={props.href} className={styles.link} onKeyUp={keyUpHandler}>
```
Comment (streetcoder123) [inline src/client/src/containers/Licenses/index.tsx]: For the Licenses file, we are going to add a `Privacy Modal`. So, these anchor tags would be converted to buttons that would open up the modal on click. Can you revert the changes for this file?

**Review by jsondoo — APPROVED** [review]

**Review by streetcoder123 — APPROVED** [review]

File: src/client/src/containers/About/index.tsx [hunk 7]
```diff
@@ -38,6 +39,7 @@ const About = ({ versions, intl }: Props) => {
         <a
           className={styles.link}
           href="https://github.com/Microsoft/WebTemplateStudio"
```
Comment (markAtMicrosoft) [inline src/client/src/containers/About/index.tsx:41]: should this be hard coded into here?  I realize it's not your code but still...  (Consider a new task to fix this, not here)

**Review by markAtMicrosoft — APPROVED** [review]
